### PR TITLE
Recycle the TLS key that holds thread_event_handler

### PR DESCRIPTION
**Pull request by zhihaoy**
fixes: #25278 

The change cleans up TLS when handling errors in `ossl_thread_event_ctx_new` as well.

### CODE REVIEW

**Review by t8m — APPROVED** [review]

File: crypto/initthread.c [hunk 1]
```diff
@@ -267,7 +267,7 @@ void *ossl_thread_event_ctx_new(OSSL_LIB_CTX *libctx)
         return NULL;
 
     if (!CRYPTO_THREAD_init_local(tlocal, NULL)) {
-        goto err;
+        goto deinit;
```
Comment (nhorman) [inline crypto/initthread.c:270]: The style checker is catching this.  I know this didn't introduce it, but coding style says that, if you have a conditional with only one statement, you don't put curly braces around it.  Removing the curly braces here will get the style check ci job to pass

**Review by nhorman — CHANGES_REQUESTED** [review]

**Review by nhorman — APPROVED** [review]

**Comment by zhihaoy** [comment]
ping

**Comment by nhorman** [comment]
Its approved, we're just waiting to merge it now, which at this point has to wait until sept 5 (alpha release), as the tree is currently frozen

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Merged to the master, 3.4, 3.3, 3.2 and 3.1 branches. Does not apply cleanly to 3.0.

Thank you for your contribution.

**Pull request closed without merging**